### PR TITLE
Add Sysmon DNS Support

### PR DESCRIPTION
**Pull request by Zalgo2462**
Adds limited DNS support by parsing Sysmon logs.

Example DNS log is attached: [dns.log](https://github.com/activecm/espy/files/8166175/dns.log)

Only the following Zeek fields are available:
- Source IP
  -  Limitation: Sysmon does not collect the source IP associated with each DNS query. However, we can collect the querying system's IP addresses and write out a record for each address.
- Query 
- Query Type ID
- Query Type Name
- Answers

These fields are enough to enable RITA's exploded DNS analysis.

Sample DNS JSON from winlogbeat: [sysmon22.txt](https://github.com/activecm/espy/files/8166353/sysmon22.txt)
Sample Conn JSON from winlogbeat: [sysmon3.txt](https://github.com/activecm/espy/files/8166362/sysmon3.txt)

### CODE REVIEW

File: espy/output/zeek/conn.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	return output, err
+}
+
+func (c ConnTSV) HandlesECSRecord(data input.ECSRecord) bool {
+	return data.Event.Provider == "Microsoft-Windows-Sysmon" && data.Event.Code == 3
+}
+
+func init() {
+	RegisteredTSVFileTypes = append(RegisteredTSVFileTypes, ConnTSV{})
+}
```
Comment (Zalgo2462) [inline espy/output/zeek/conn.go:97]: When the package is imported, `RegisteredTSVFileTypes` is set to `{ConnTSV{}, DnsTSV{}}`. This provides a simple way for the file writers to query which types of Zeek files are supported and map ECS records to each of the files.

File: espy/output/zeek/rolling.go [hunk 2]
```diff
@@ -24,31 +27,45 @@ const rotateOnMinute = false
...
 type RollingWriter struct {
-	zeekDir     string
-	spoolDir    string
-	spoolFile   string
+	archiveDir string
+	spoolDir   string
+
+	fs         afero.Fs
+	clock      clock.Clock
+	spoolFiles map[TSVFileType]afero.File
```
[the remark this answers is not in the document]
Comment (Zalgo2462) [reply espy/output/zeek/rolling.go:35]: These will likely be the most difficult changes to review. I've included updated golang tests to help with the process.

File: espy/output/zeek/dns.go [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+		if err != nil {
+			return output, input.ErrMalformedECSRecord
+		}
+
+		answersSetBuilder := strings.Builder{}
+		answerTypeName := header.UnsetField
+		answerTypeID := header.UnsetField
+
+		// Don't report CNAME responses since Zeek doesn't
+		// WEIRD: Windows issues A queries for IP addresses and gets back "-" in the answers
```
Comment (Zalgo2462) [inline espy/output/zeek/dns.go:138]: Example DNS query where this is an issue: [sysmon22weird.txt](https://github.com/activecm/espy/files/8166364/sysmon22weird.txt)

**Review by fullmetalcache — APPROVED** [review]
Nothing jumps out at me as needing to be changed.

Tested on Win10, Server2019, and Server2022 and everything looks like it's working.

Nice work!

**Pull request merged**